src/components/discover-sheet/PulseIndexSection.js

```diff
@@ -1,25 +1,209 @@
-import React from 'react';
-import { lightModeThemeColors } from '../../styles/colors';
+import { get } from 'lodash';
+import React, { Fragment, useCallback, useMemo } from 'react';
+import { View } from 'react-native';
+import LinearGradient from 'react-native-linear-gradient';
+import { useSelector } from 'react-redux';
+import { DPI_ADDRESS } from '../../references/indexes';
+import font from '../../styles/fonts';
+import { ButtonPressAnimation } from '../animations';
+import { CoinIcon } from '../coin-icon';
+import { Column, Row } from '../layout';
 import { Text } from '../text';
+import ChartTypes from '@rainbow-me/helpers/chartTypes';
+import { useAccountSettings } from '@rainbow-me/hooks';
+import { useNavigation } from '@rainbow-me/navigation';
+import Routes from '@rainbow-me/routes';
+import { fontWithWidth, position } from '@rainbow-me/styles';
+import { handleSignificantDecimals } from '@rainbow-me/utilities';
+import { ethereumUtils } from '@rainbow-me/utils';
 import ShadowStack from 'react-native-shadow-stack';
 
-export const PulseIndexShadow = [
-  [0, 2, 5, lightModeThemeColors.orange, 0.2],
-  [0, 6, 10, lightModeThemeColors.orange, 0.14],
-  [0, 1, 18, lightModeThemeColors.orange, 0.12],
-];
+const formatItem = ({ address, name, price, symbol }, nativeCurrencySymbol) => {
+  const change = `${parseFloat(
+    (price?.relative_change_24h || 0).toFixed(2)
+  )}%`.replace('-', '');
+
+  const value = `${nativeCurrencySymbol}${handleSignificantDecimals(
+    price?.value,
+    2
+  )} `;
+
+  return {
+    address,
+    change,
+    isPositive: price?.relative_change_24h > 0,
+    name,
+    price: value,
+    symbol,
+  };
+};
+
+const PulseIndex = () => {
+  const { navigate } = useNavigation();
+  const { genericAssets } = useSelector(({ data: { genericAssets } }) => ({
+    genericAssets,
+  }));
+
+  const charts = useSelector(({ charts: { charts } }) => charts);
+
+  const chartPriceDataForDPI = useMemo(
```

Comment:
Given that this is essentially an O(1) operation, we don't really need the useMemo here

Reply:
I think the parsing that the second parameter of the get is doing is not free, but I'll remove the memo anyway